### ISSUE TITLE
Update on V2 release and checklist

### ISSUE THREAD

**Issue by bawahakim** [flyerhq/flutter_chat_ui#592]
Hello, first thank you for this library, very flexible and just works out of the box! Can see a lot of effort and attention has been put in.

For a few months, I've seen that V2 is in the work, but no estimate yet. Would it be possible to have tentative release date? 😄 

Is there a checklist of things missing for V2? Maybe the community could help out?

Thanks again!

**Comment by demchenkoalex** [comment]
hey, thank you as well!

I almost had `v2` done already, where I met a blocker - lack of an appropriate caching package. My plan with `v2` is killer performance and this is not achievable without caching. My plan was to use https://github.com/Baseflow/flutter_cache_manager, but with it I lack control I need to achieve beautiful, performant UI. I tried searching for alternative packages, but there are none. So currently I am working on another big project, caching package, that I need to finish `v2`. I did not plan for this and turned out writing such a package is not easy as well. So I continue my struggle :/ Unfortunately I am alone, with full time job and normal life with PTOs, with no additional help (some of community members reached out to help, but nobody really committed so far) so we have what we have

**Comment by demchenkoalex** [comment]
everything I have is on a `v2` branch (except the cache) will try to have something simple first to unblock the chat and then improve (maybe with the community help)

**Comment by petrnymsa** [comment]
Hi, @demchenkoalex, we are considering using this awesome-looking package in our app. However, I am pretty worried about migration from V1 to V2. 

Is there any estimate of when V2 could land? (We can postpone the implementation, but we need to know the estimated date). 
If we start implementing our chat in V1, how difficult will it be to migrate to V2? 

Thanks!

**Comment by demchenkoalex** [comment]
Hey, I finally solved the problem I had for months and will create a v2 tracking issue soon, that will be used to track progress. Currently v2 does not even have message input and it lacks even basic features, but solid foundation is there. My plan is to build on top of the foundation and allow contributions as well, so I hope things will go quickly from here. Hopefully I will not meet new blockers along the way 🤞

**Comment by CHPisces** [comment]
I am also using this great library, first of all thank you to the author, and then want to ask if there are any plans to update the v2 version now? I am experiencing performance issues with the v1 version and may consider upgrading.

**Comment by demchenkoalex** [comment]
Hey, `v2` is very close to alpha release now, however it might lack some features from original release, but I hope I will be able to quickly add everything important and then build new features on top

**Comment by CHPisces** [comment]
> Hey, `v2` is very close to alpha release now, however it might lack some features from original release, but I hope I will be able to quickly add everything important and then build new features on top

How much work does it take to upgrade from v1 version to v2 version

**Comment by demchenkoalex** [comment]
There will be no guide as I just throw everything out and created everything from scratch. That being said I will try to keep models as similar as possible, so if you don't have much customisation it will be just rename of a few props. If there is a customisation, new version might make it easier (for example providing custom font is now 1 line and not new definition of a whole theme etc.)

**Comment by demchenkoalex** [comment]
update is here https://github.com/flyerhq/flutter_chat_ui/issues/646
